src/os/bluestore/BlueStore.cc

```diff
@@ -6409,6 +6418,7 @@ void BlueStore::_txc_finish_io(TransContext *txc)
     if (p->state < TransContext::STATE_IO_DONE) {
       dout(20) << __func__ << " " << txc << " blocked by " << &*p << " "
 	       << p->get_state_name() << dendl;
+      txc->is_pipelined_io = true;
```

Comment:
Umm... is_pipelined_io flag is used for checking if ShardedOpWQ holding pg_lock is running
If i change where is_piplined_io is set as you can suggested(STATE_PREPARE),  then i can remove it.
Then, my intention is that is_pipelined_io is set to check if this code line is executed by ShardedOpWQ. This patch already can check any of the txc's ahead of us by osr->kv_finisher_submitting counter.

Thank you